gnucash/gnome-utils/gnc-autoclear.cpp

```diff
@@ -0,0 +1,198 @@
+/********************************************************************
+ * gnc-autoclear.c -- Knapsack algorithm functions                  *
+ *                                                                  *
+ * Copyright 2020 Cristian Klein <cristian@kleinlabs.eu>            *
+ * Modified  2021 Christopher Lam to clear same-amount splits       *
+ * Modified  2025 ChatGPT to merge without building a shifted vec   *
+ *                                                                  *
+ * This program is free software; you can redistribute it and/or    *
+ * modify it under the terms of the GNU General Public License as   *
+ * published by the Free Software Foundation; either version 2 of   *
+ * the License, or (at your option) any later version.              *
+ *                                                                  *
+ * This program is distributed in the hope that it will be useful,  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    *
+ * GNU General Public License for more details.                     *
+ *                                                                  *
+ * You should have received a copy of the GNU General Public License*
+ * along with this program; if not, contact:                        *
+ *                                                                  *
+ * Free Software Foundation           Voice:  +1-617-542-5942       *
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       *
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
+ *******************************************************************/
+#include <config.h>
+
+#include <gtk/gtk.h>
+#include <glib/gi18n.h>
+
+#include "Account.h"
+#include "Account.hpp"
+#include "Split.h"
+#include "Transaction.h"
+#include "gnc-autoclear.h"
+
+#include <vector>
+#include <numeric>
+#include <algorithm>
+#include <cstdint>
+
+#define log_module "autoclear"
```

Comment:
This should be
```
static QofLogModule log_module = "gnc.autoclear"
```
`gnc` is important, it allows adjusting the logging level for the whole application.

On the topic of logging, this module should use more, at least `ENTER()` and `LEAVE()` macros on the major functions.